pkg/network/ebpf/c/protocols/kafka/helpers.h

```diff
@@ -0,0 +1,298 @@
+        READ_BIG_ENDIAN_WRAPPER(s16, transactional_id_size, skb, offset);
+        if (transactional_id_size > 0) {
+            offset += transactional_id_size;
+        }
+    }
+
+    READ_BIG_ENDIAN_WRAPPER(s16, acks, skb, offset);
+    if (acks > 1 || acks < -1) {
+        // The number of acknowledgments the producer requires the leader to have received before considering a request
+        // complete. Allowed values: 0 for no acknowledgments, 1 for only the leader and -1 for the full ISR.
+        return false;
+    }
+
+    READ_BIG_ENDIAN_WRAPPER(s32, timeout_ms, skb, offset);
+    if (timeout_ms < 0) {
+        // timeout_ms cannot be negative.
+        return false;
+    }
+
+    return validate_first_topic_name(skb, offset);
+}
+
+// Trying to parse a fetch request.
+static __always_inline bool try_parse_fetch_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    // replica_id => INT32
+    // max_wait_ms => INT32
+    // min_bytes => INT32
+    offset += 3 * sizeof(s32);
+
+    if (kafka_header->api_version >= 7) {
+        // session_id => INT32
+        // session_epoch => INT32
+        offset += 2 * sizeof(s32);
+    }
+
+    if (kafka_header->api_version >= 4) {
+        // isolation_level => INT8
+        offset += sizeof(s8);
+    }
+
+    if (kafka_header->api_version >= 3) {
+        // max_bytes => INT32
+        offset += sizeof(s32);
+    }
+
+    return validate_first_topic_name(skb, offset);
+}
+
+// Calls the relevant function, according to the api_key.
+static __always_inline bool is_kafka_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    switch (kafka_header->api_key) {
+    case KAFKA_PRODUCE:
+        return try_parse_produce_request(kafka_header, skb, offset);
+    case KAFKA_FETCH:
+        return try_parse_fetch_request(kafka_header, skb, offset);
+    default:
+        return false;
+    }
+}
+
+// Checks if the packet represents kafka communication.
+static __always_inline bool is_kafka(struct __sk_buff *skb, skb_info_t *skb_info, const char* buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, KAFKA_MIN_LENGTH);
+
+    const kafka_header_t *header_view = (kafka_header_t *)buf;
+    kafka_header_t kafka_header;
+    bpf_memset(&kafka_header, 0, sizeof(kafka_header));
+    kafka_header.message_size = bpf_ntohl(header_view->message_size);
+    kafka_header.api_key = bpf_ntohs(header_view->api_key);
+    kafka_header.api_version = bpf_ntohs(header_view->api_version);
+    kafka_header.correlation_id = bpf_ntohl(header_view->correlation_id);
+    kafka_header.client_id_size = bpf_ntohs(header_view->client_id_size);
+
+    // Checking if the header is valid.
+    if (!is_valid_kafka_request_header(&kafka_header)) {
+        return false;
+    }
+
+    u32 offset = skb_info->data_off + sizeof(kafka_header_t);
+    // Validate client ID
+    if (!is_valid_client_id(skb, offset, kafka_header.client_id_size)) {
```

Comment:
Client id is part of the Kafka header according to the protocol, consider calling it from `is_valid_kafka_request_header`

Reply:
no, the reason for splitting them is to avoid passing offset as an out parameter